main.md

```diff
@@ -262,6 +262,82 @@ other client authentication methods.
 DPoP does not directly ensure message integrity but relies on the TLS
 layer for that purpose. See (#Security) for details.
 
+# Public-Key PKCE
+
+PKCE allows a client to prove that the sender of a token request was also the
+sender of a prior authorization request {{!RFC7636}}.  Existing PKCE methods are
+based on symmetric cryptography.  In this section, we specify a new PKCE method
+"ECVR" based on asymmetric cryptography. 
+
+The "ES256" PKCE method presumes that a client holds an ECDSA key pair with
+private key `pkce_key_priv` and public key `pkce_key_pub`.  The client also
+generates a fresh 32-byte random value `pkce_random` for each PKCE challenge.
+The PKCE code challenge commits to these values by sending the JWK thumbprint of
+`pkce_key_pub` {{!RFC7638}} and the SHA-256 hash of `pkce_random`.
+
```

Comment:
I think DPoP allows any algorithms defined in https://datatracker.ietf.org/doc/html/rfc7518, so this PKCE method may need to be similarly flexible to make it compatible with DPoP?